### PR TITLE
chore: improve grid render performance

### PR DESCRIPTION
**Pull request by bluestreak01**
## Motivation

To prepare grid for column movement and column freezing I moved grid render from stylesheets to element styles. This in itself improved grid scrolling performance.

### Before

image

### After

image

## Look and feel changes

- added subtle vertical lines to separate cell values
- removed cell hover to avoid ambiguously highlighting multiple grid cells
- made row hover background more subdued
- there is border on the right edge of the grid to make look and feel consistent

image

### CODE REVIEW

File: packages/web-console/src/js/console/grid.js [hunk 1]
```diff
@@ -603,12 +572,12 @@ export function grid(root, msgBus) {
...
 
   function removeClass(e, className) {
-    e.className = e.className.replace(className, '')
+    e.className = e.className.replace(className, '').trim()
   }
 
   function addClass(e, className) {
     if (e && !e.className.includes(className)) {
-      e.className += className
+      e.className += ' ' + className
```
Comment (argshook) [inline packages/web-console/src/js/console/grid.js:580]: no need to change anything, just want to share info about `classlist`, might be helpful for lines like this.

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

**Review by argshook — APPROVED** [review]
This is awesome!

File: packages/web-console/src/js/console/grid.js (hunk 1 above)
Comment (bluestreak01) [reply packages/web-console/src/js/console/grid.js:577]: hmm, do that specifically did not work for me, let me retry

Comment (bluestreak01) [reply packages/web-console/src/js/console/grid.js:577]: ah, `e.classList.add()` - thanks!

Comment (bluestreak01) [reply packages/web-console/src/js/console/grid.js]: addressed in https://github.com/questdb/ui/pull/86/commits/9c85caacdeda3c70a6465cdb5caab1b9df8f4565

**Comment by bluestreak01** [comment]
also highlighted nulls

image

File: packages/web-console/src/styles/_grid.scss [hunk 2]
```diff
@@ -113,10 +131,11 @@ $visible-drag-handle-margin: 10px;
 .qg-drag-handle {
   // place drag handle so that visible "line" is in the middle of it
   // to add hysteresis for the drag cursor
-  margin-left: -$visible-drag-handle-margin / 2;
-  width: $visible-drag-handle-margin * 2 !important;
+  margin-left: -$visible-drag-handle-margin - $drag-handle-margin;
+  width: $visible-drag-handle-margin * 2 ;
   height: $row-height;
```
Comment (argshook) [inline packages/web-console/src/styles/_grid.scss:136]:
```suggestion
  width: $visible-drag-handle-margin * 2;
  height: 100%;
  top: 0;
```

**Review by argshook — CHANGES_REQUESTED** [review]

File: packages/web-console/src/styles/_grid.scss (hunk 2 above)
Comment (argshook) [reply packages/web-console/src/styles/_grid.scss:136]: without this change, the drag handle position is off:

Screenshot 2023-02-13 at 15 12 17

**Pull request merged**